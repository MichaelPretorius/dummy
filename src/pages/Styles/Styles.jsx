import React, { useState } from 'react';
import { Link } from 'react-router-dom';

import logo from './communicate.svg';
import burger from './navburger.png';
import {
  Container,
  Wrapper,
  Header,
  Card,
  Display,
  Title,
  Body,
  Headline,
  Caption,
  GroupButton,
  LabelButton,
  Button,
  InputField,
  Label,
  Navbar,
  NavBrand,
  Navs,
  Nav,
  NavContainer,
  Burger,
  LinkButton,
  FloatingActionButton,
} from '../../styles';

const Styles = () => {
  const [showNav, setShowNav] = useState(false);

  const toggleNav = () => {
    setShowNav(!showNav);
  };

  return (
    <>
      {/* navbar.js */}
      <NavContainer row secondary>
        <Wrapper row>
          <Burger onClick={toggleNav} src={burger} />
          <NavBrand to="/">yDS-template</NavBrand>
        </Wrapper>
        <Navbar primary showNav={showNav} width="200px">
          <Navs>
            <Nav to="/">link</Nav>
            <Nav to="/">link</Nav>
            <Nav to="/">link</Nav>
          </Navs>
        </Navbar>
      </NavContainer>

      <Container>
        {/* typography.js */}
        <Header primary>Typography:</Header>
        <Display>Display Fonts for Modal Titles</Display>
        <Header secondary>Header for Page Titles</Header>
        <Title center>Title for Tabs, Titles and Forms</Title>
        <Headline>Headline for Info Paragraphs!</Headline>
        <Body>Body for descriptions</Body>
        <Caption right>
          Caption for timestamps, footers and additional info
        </Caption>

        {/* cards.js */}
        <Header primary>Cards:</Header>
        <Card fill row primary>
          <Wrapper alignCenter>
            <Display>fill</Display>
            <img src={logo} alt="logo" />
          </Wrapper>
          <Body>
            Lorem ipsum dolor amet cronut vape kickstarter, farm-to-table
            everyday carry swag synth man braid chia mumblecore post-ironic
            gastropub locavore vegan poke. Hell of gochujang subway tile
            fashion.
          </Body>
          <Wrapper justifyCenter>
            <Button secondary small round>
              small Button
            </Button>
            <Button small round>
              small Button
            </Button>
          </Wrapper>
        </Card>
        <Card>
          <img src={logo} alt="logo" />
          <Display>Media Cards</Display>
          <Body>
            Lorem ipsum dolor amet cronut vape kickstarter, farm-to-table
            everyday carry swag synth man braid chia mumblecore post-ironic
            gastropub locavore vegan poke. Hell of gochujang subway tile fashion
            axe flannel, cred williamsburg coloring book. Blog health goth
            succulents fam, fixie sartorial raclette glossier. Squid seitan
            chambray neutra live-edge iPhone taiyaki.
          </Body>
          <Caption right>7 days ago</Caption>
          <Wrapper row>
            <Button primary small round>
              small Button
            </Button>
            <Button small round>
              small Button
            </Button>
          </Wrapper>
          <Button secondary round fill>
            round fill
          </Button>
        </Card>

        {/* buttons.js */}
        <Header primary>Buttons:</Header>
        <Wrapper row>
          <Button primary>primary</Button>
          <Button secondary round>
            secondary round
          </Button>
        </Wrapper>
        <Button small round right>
          small round right
        </Button>
        <Wrapper>
          <LinkButton primary round to="/">
            LinkButton round
          </LinkButton>
          <Button secondary small submit>
            small submit
          </Button>
        </Wrapper>
        <Wrapper row end>
          <FloatingActionButton as={Link} to="/">
            Link
          </FloatingActionButton>
          <LabelButton primary>label button</LabelButton>
        </Wrapper>
        <GroupButton primary>Group</GroupButton>
        <GroupButton secondary>Group</GroupButton>
        <Button secondary fill round>
          fill
        </Button>

        {/* formFields.js */}
        <Header primary>Forms:</Header>
        <form>
          <Label>
            text transition fill
            <InputField transition fill />
          </Label>
          <Wrapper row between wrap>
            <Label primary>
              small
              <InputField small transition />
            </Label>
            <Label>
              primary
              <InputField primary />
            </Label>
          </Wrapper>
        </form>
      </Container>
    </>
  );
};

export default Styles;
