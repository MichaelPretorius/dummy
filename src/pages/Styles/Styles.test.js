import React from 'react';
import { shallow } from 'enzyme';

import { Styles } from './Styles';

describe('Styles component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Styles />);
  });

  test('should render Styles component', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
