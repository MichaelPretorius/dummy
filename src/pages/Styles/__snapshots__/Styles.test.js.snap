// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styles component should render Styles component 1`] = `
"<Fragment>
  <Header primary={true}>
    Typography:
  </Header>
  <Display>
    Display Fonts for Modal Titles
  </Display>
  <Header secondary={true}>
    Header for Page Titles
  </Header>
  <Title center={true}>
    Title for Tabs, Titles and Forms
  </Title>
  <Headline>
    Headline for Info Paragraphs!
  </Headline>
  <Body>
    Body for descriptions
  </Body>
  <Caption right={true}>
    Caption for timestamps, footers and additional info
  </Caption>
  <Header primary={true}>
    Buttons:
  </Header>
  <Wrapper row={true} justifyCenter={true}>
    <Button primary={true}>
      primary
    </Button>
    <Button secondary={true} round={true}>
      secondary round
    </Button>
  </Wrapper>
  <Button small={true} round={true} right={true}>
    small round right
  </Button>
  <Wrapper width=\\"30%\\">
    <LinkButton primary={true} round={true} to=\\"/\\">
      LinkButton round
    </LinkButton>
    <Button secondary={true} small={true} submit={true}>
      small submit
    </Button>
  </Wrapper>
  <Wrapper row={true} justifyEnd={true}>
    <FloatingActionButton as={{...}} to=\\"/\\">
      Link
    </FloatingActionButton>
    <LabelButton primary={true}>
      label button
    </LabelButton>
  </Wrapper>
  <GroupButton primary={true}>
    Group
  </GroupButton>
  <GroupButton secondary={true}>
    Group
  </GroupButton>
  <Button secondary={true} fill={true} round={true}>
    fill
  </Button>
  <Header primary={true}>
    Forms:
  </Header>
  <Form>
    <Label>
      text transition width
      <InputField transition={true} width=\\"100%\\" />
    </Label>
    <Wrapper row={true} justifyBetween={true} wrap={true}>
      <Label primary={true}>
        small
        <InputField small={true} transition={true} />
      </Label>
      <Label>
        primary
        <InputField primary={true} />
      </Label>
    </Wrapper>
  </Form>
  <Header primary={true}>
    Cards:
  </Header>
  <Card fill={true} row={true} primary={true} alignCenter={true}>
    <Wrapper alignCenter={true}>
      <Display>
        fill
      </Display>
      <img src=\\"communicate.svg\\" alt=\\"logo\\" />
    </Wrapper>
    <Body>
      Lorem ipsum dolor amet cronut vape kickstarter, farm-to-table everyday carry swag synth man braid chia mumblecore post-ironic gastropub locavore vegan poke. Hell of gochujang subway tile fashion.
    </Body>
    <Wrapper>
      <Button secondary={true} small={true} round={true}>
        small Button
      </Button>
      <Button small={true} round={true}>
        small Button
      </Button>
    </Wrapper>
  </Card>
  <Card>
    <img src=\\"communicate.svg\\" alt=\\"logo\\" />
    <Display>
      Media Cards
    </Display>
    <Body>
      Lorem ipsum dolor amet cronut vape kickstarter, farm-to-table everyday carry swag synth man braid chia mumblecore post-ironic gastropub locavore vegan poke. Hell of gochujang subway tile fashion axe flannel, cred williamsburg coloring book. Blog health goth succulents fam, fixie sartorial raclette glossier. Squid seitan chambray neutra live-edge iPhone taiyaki.
    </Body>
    <Caption right={true}>
      7 days ago
    </Caption>
    <Wrapper row={true} justifyEvenly={true}>
      <Button primary={true} small={true} round={true}>
        small Button
      </Button>
      <Button small={true} round={true}>
        small Button
      </Button>
    </Wrapper>
    <Button secondary={true} round={true} fill={true}>
      round fill
    </Button>
  </Card>
</Fragment>"
`;
